### PR TITLE
Task sf/j150 save recored episode functionality

### PR DESCRIPTION
**Pull request by Sayonarko**

### CODE REVIEW

File: frontend/src/services/record-audio/record-audio.service.ts [hunk 1]
```diff
@@ -28,7 +27,7 @@ class RecordAudio {
   }
 
   public start(): void {
-    this.#recorder?.start();
+    this.#recorder?.start(500);
```
Comment (Denidzio) [inline frontend/src/services/record-audio/record-audio.service.ts]: What does it meen 500?

Comment (Sayonarko) [reply frontend/src/services/record-audio/record-audio.service.ts]: This parameter takes a value in milliseconds, and overrides the length of the media capture chunk to be returned in each Blob.

File: frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx [hunk 2]
```diff
@@ -57,14 +60,16 @@ const CreateEpisodeForm: React.FC<Props> = ({ onSubmit, payload = DEFAULT_CREATE
...
-        <label className={styles.recordLabel}>
-          Upload record
-          <input
-            {...register(EpisodePayloadKey.RECORD)}
-            accept={acceptAudioExtension}
-            type={InputType.FILE}
-          />
-        </label>
+        {hasRecord
+          || <label className={styles.recordLabel}>
```
Comment (what1s1ove) [inline frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx]:
```suggestion
        {hasRecord
          && <label className={styles.recordLabel}>
```

File: frontend/src/services/record-audio/record-audio.service.ts (hunk 1 above)
Comment (JuliaDzyuba) [reply frontend/src/services/record-audio/record-audio.service.ts]: Magic number? it should be constant?

**Review by JuliaDzyuba — DISMISSED** [review]

File: frontend/src/services/record-audio/record-audio.service.ts (hunk 1 above)
Comment (Sayonarko) [reply frontend/src/services/record-audio/record-audio.service.ts]: You right. It should be constant

**Review by Denidzio — DISMISSED** [review]

**Review by artem-totality — DISMISSED** [review]

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx (hunk 2 above)
Comment (Sayonarko) [reply frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx]: If we have a record, then we hide the upload record button

File: frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx [hunk 3]
```diff
@@ -52,11 +53,13 @@ const CreateEpisodeForm: React.FC<Props> = ({ onSubmit, payload = DEFAULT_CREATE
           errors={errors}
           imageUrl={imageUrl}
         />
-        <RecordPreviewControl
-          name={EpisodePayloadKey.RECORD}
-          control={control}
-          errors={errors}
-        />
+        {hasLiveRecord
+          || <RecordPreviewControl
```
Comment (what1s1ove) [inline frontend/src/components/configurate-episode/components/create-episode-form/create-episode-form.tsx]:
```suggestion
          && <RecordPreviewControl
```
🤔

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by EugeniusGR — DISMISSED** [review]

**Review by EugeniusGR — DISMISSED** [review]

**Review by yaroslav207 — DISMISSED** [review]

File: frontend/src/components/configurate-episode/components/create-episode-form/components/record-preview-control/record-preview-control.tsx [hunk 4]
```diff
@@ -28,16 +30,22 @@ const RecordPreviewControl: React.FC<Props> = ({
...
   errors,
+  hasLiveRecord,
 }) => {
   const { field } = useController({ name, control });
 
   const [currentRecord, setRecord] = useState<string | undefined>('');
 
+  const liveRecordName = hasLiveRecord
+    ? getFormattedDate(new Date(), DateFormatType.DAY_MONTH_YEAR) + '.wav'
+    : '';
```
Comment (what1s1ove) [inline frontend/src/components/configurate-episode/components/create-episode-form/components/record-preview-control/record-preview-control.tsx]: we already have enum for file extension. Use it instead

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/configurate-episode/components/create-episode-form/components/record-preview-control/record-preview-control.tsx [hunk 5]
```diff
@@ -28,16 +30,22 @@ const RecordPreviewControl: React.FC<Props> = ({
...
   errors,
+  hasLiveRecord,
 }) => {
   const { field } = useController({ name, control });
 
   const [currentRecord, setRecord] = useState<string>('');
 
+  const liveRecordName = hasLiveRecord
+    ? `${getFormattedDate(new Date(), DateFormatType.DAY_MONTH_YEAR)}.${FileExtension.WAV}`
+    : '';
```
Comment (what1s1ove) [inline frontend/src/components/configurate-episode/components/create-episode-form/components/record-preview-control/record-preview-control.tsx]: create a `getLiveRecorName` helper and move it to frontend/src/components/configurate-episode/components/create-episode-form/components/record-preview-control/helpers/
```suggestion
  const liveRecordName = hasLiveRecord
    ? getLiveRecorName()
    : '';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**